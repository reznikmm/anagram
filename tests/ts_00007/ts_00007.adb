--  Check extended attributed grammar reader

with Anagram.Grammars.Reader;
with Anagram.Grammars_Debug;

procedure TS_00007 is
   G : constant Anagram.Grammars.Grammar :=
     Anagram.Grammars.Reader.Read ("test.ag");
begin
   Anagram.Grammars_Debug.Print (G);
end TS_00007;
