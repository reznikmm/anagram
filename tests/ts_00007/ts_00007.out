Terminals:
 1 T
  Attr: ast inherited=FALSE
Non Terminals:
 1 NT
  Attr: ast inherited=FALSE
  Attr: env inherited=TRUE
  Production: NT-T
    T refs T
    NT2 refs NT2
Rule: 
  Result: LHS.ast
  Arg: T.ast
Rule: 
  Result: NT2.env
  Arg: LHS.env
  Production: NT-TT
    t1 refs T
    t2 refs T
  Production: NT-TList
    tlist List:
  Production: NT-TList-tlist-
    T refs T
    End
  Production: NT-TOpt
     Option:
  Production: NT-TOpt-
    T refs T
    nt2 refs NT
    End
 2 NT2
  Attr: ast inherited=FALSE
  Attr: env inherited=TRUE
  Production: NT2-
    T refs T
